src/Mod/BIM/ArchStairs.py

```diff
@@ -104,7 +104,9 @@ def setProperties(self,obj):
         if not "TreadThickness" in pl:
             obj.addProperty("App::PropertyLength","TreadThickness","Steps",QT_TRANSLATE_NOOP("App::Property","The thickness of the treads"), locked=True)
         if not "BlondelRatio" in pl:
-            obj.addProperty("App::PropertyFloat","BlondelRatio","Steps",QT_TRANSLATE_NOOP("App::Property","The Blondel ratio indicates comfortable stairs and should be between 62 and 64cm or 24.5 and 25.5in"), locked=True)
+            # Change from unitless float to Length
+            #obj.addProperty("App::PropertyFloat","BlondelRatio","Steps",QT_TRANSLATE_NOOP("App::Property","The Blondel ratio indicates comfortable stairs and should be between 62 and 64cm or 24.5 and 25.5in"), locked=True)
+            obj.addProperty("App::PropertyLength","BlondelRatio","Steps",QT_TRANSLATE_NOOP("App::Property","The Blondel ratio indicates comfortable stairs and should be between 62 and 64cm or 24.5 and 25.5in"), locked=True)
```

Comment:
This change requires code to update older files. The old property must be removed from old stairs, the new property added and the ratio calculated.

Reply:
Thanks for the comments.

I thought we do not need to update the old model as there is no other foreseeable code which would use this.  In fact, I think virtually there is scarce of people who would actually bother to read this.  And there have been a long period the ratio was not calculated, seems no strong reaction until this one.

Any other thought?

Reply:
Removing property which is added 'locked' return False:
`                obj.removeProperty("BlondelRatio")`

Any idea how this could be removed in the first place? 

Thanks.

>             if "BlondelRatio" in pl:
>                 obj.removeProperty("BlondelRatio")
>             # Change from unitless float to Length
>             obj.addProperty("App::PropertyLength","BlondelRatio","Steps",QT_TRANSLATE_NOOP("App::Property","The Blondel ratio indicates comfortable stairs and should be between 62 and 64cm or 24.5 and 25.5in"), locked=True)
>

Reply:
Ok, hope this works, found this:
-  https://github.com/FreeCAD/FreeCAD/pull/20778#issuecomment-3365156656


“ To remove the lock :
joint.setPropertyStatus("Distance2", "-LockDynamic") “